File: app/src/main/java/test/juyoufuli/com/myapplication/app/ui/tab/adapter/SystemDataAdapter.java
package test.juyoufuli.com.myapplication.app.ui.tab.adapter;

import android.view.View;

import com.jess.arms.base.BaseHolder;
import com.jess.arms.base.DefaultAdapter;

import java.util.List;

import test.juyoufuli.com.myapplication.R;
import test.juyoufuli.com.myapplication.mvp.entity.Datas;
import test.juyoufuli.com.myapplication.mvp.entity.SystemBean;

/**
 * Author : ludf
 * Created Time : 2018-09-27  14:57
 * Description:
 */
public class SystemDataAdapter extends DefaultAdapter<SystemBean> {
    public SystemDataAdapter(List<SystemBean> infos) {
        super(infos);
    }

    @Override
    public BaseHolder<SystemBean> getHolder(View v, int viewType) {
        return new SystemDataItemHolder(v);
    }

    @Override
    public int getLayoutId(int viewType) {
        return R.layout.article_item;
    }
}

File: app/src/main/java/test/juyoufuli/com/myapplication/mvp/ui/system/adapter/SystemDataAdapter.kt
package test.juyoufuli.com.myapplication.mvp.ui.system.adapter

import android.content.Context
import test.juyoufuli.com.myapplication.R
import test.juyoufuli.com.myapplication.app.recyclerview.CommonAdapter
import test.juyoufuli.com.myapplication.app.recyclerview.base.ViewHolder
import test.juyoufuli.com.myapplication.databinding.ArticleSystemItemBinding
import test.juyoufuli.com.myapplication.mvp.entity.SystemBean

/**
 * Author : ludf
 * Created Time : 2018-09-27  14:57
 * Description:
 */
class SystemDataAdapter(var context: Context, infos: List<SystemBean>) :
    CommonAdapter<SystemBean>(context, R.layout.article_system_item, infos) {


    override fun convert(holder: ViewHolder, t: SystemBean, position: Int) {
        val bind = ArticleSystemItemBinding.bind(holder.itemView)
        bind.tvChapterName.text = t.name
        var stringName = ""
        t.children.forEach {
            stringName += it.name + ","
        }
        bind.tvDesc.text = stringName
    }
}
